### DISCUSSION TITLE
Forelse Loop giving error "nextSibling Error"

### DISCUSSION THREAD

**Discussion by Rashuoo7** [livewire/livewire#6674] [1. Help] [answered]
I just upgraded to Livewire 3 and application is giving errors. I am getting nextSibling error after any function runs.
The Code is below : 
@forelse($courses as $course)
                <x-table.row >
                    <x-table.cell>
                        <div class="text-sm font-medium text-gray-900">
                            <a href="{{route('admin.courses.show', ['course' => $course->id])}}">
                                {{ $course->title }}
                            </a>
                        </div>
                    </x-table.cell>
                    <x-table.cell>
                    {{--Edit Course Button--}}
                    <div x-data="{ open:false }" class="w-20 font-medium">
                        <a wire:click="edit({{$course->id}})"  href="#" @click="open = !open" class="text-yellow-500 hover:text-yellow-600">Edit</a>
                    </div>
                        {{-- /Edit Course Button--}}
                    </x-table.cell>
                    <x-table.cell>
                        <a href="#" wire:click="delete({{$course->id}})" class=" font-medium text-red-600 hover:text-red-900">Delete</a>
                    </x-table.cell>
                </x-table.row>
            @empty
                <x-table.row>
                    <x-table.empty_cell colspan="16"></x-table.empty_cell>
                </x-table.row>
            @endforelse
![image](https://github.com/livewire/livewire/assets/22994814/71e83437-2e0c-4738-b968-4e4cb92134ec)

**Comment by addtobag** [comment]
What version of Livewire are you on? Should be fixed in 3.0.1 that was released last week.

**Reply by Rashuoo7** [reply]
I am using Livewire 3.0

**Reply by addtobag** [reply]
Okay you need to be on 3.0.1, you can update this in your composer.json file, and then by running composer update from your cli

**Reply by Rashuoo7** [reply]
Using this config in composer.json file.     "livewire/livewire": "3.0",

**Reply by addtobag** [reply, marked as answer]
Needs to be: 

"livewire/livewire": "^3.0.1",

then run 

composer update in your cli

**Reply by Rashuoo7** [reply]
I have one more issue : Is this dot notation with wire:model doen't work in livewire 3 ? wire:model="editing.title"

**Reply by addtobag** [reply]
Can you share your component code?

**Reply by Rashuoo7** [reply]
`<?php

namespace App\Livewire;

use App\Models\Course;
use Livewire\Component;

class CoursesAdmin extends Component
{
    public string $name = '';

    public $deleting;

    public string $title = '';

    public $course;

    public Course $editing;

    public $showCreateModal = false;

    public $showEditModal = false;

    public $showById = null;

    protected $rules = [
        'editing.title' => 'required',
    ];

    /**
     * @var true
     */
    public $showDeleteModal;

    public function render()
    {
        return view('livewire.courses-admin')
            ->with('courses', Course::get(['title','id']));
    }

    public function create()
    {
        $this->showCreateModal = true;
    }

    public function store()
    {
        $data = $this->validate([
            'title' => 'required',
        ]);

        $res = Course::create([
            'title' => $data['title'],
        ]);
        $this->showCreateModal = false;
    }

    public function edit(Course $course): void
    {
        $this->editing = $course;
//        dd($this->editing);
        $this->showEditModal = true;
    }

    public function update()
    {
        $this->validate();
        $this->editing->save();
        $this->showEditModal = false;
    }

    public function delete(Course $course)
    {
        $this->deleting = $course;
        $this->showDeleteModal = true;
    }

    public function destroy()
    {
        $this->deleting->delete();

        $this->showDeleteModal = false;
    }
}
`

**Reply by addtobag** [reply]
https://livewire.laravel.com/docs/upgrading#eloquent-model-binding

Read this part of the upgrade guide.

You can set 'legacy_model_binding' => true, in your config/livewire.php file to bring it back.

Alternatively, you can look at creating a form object:
https://livewire.laravel.com/docs/forms#extracting-a-form-object

**Reply by Rashuoo7** [reply]
Thank You Caleb!

**Comment by devhau** [comment]
you need add wire:key for `<x-table.row > => <x-table.row wire:key="row-{{$course->id}}">`
and 
```
@empty
<x-table.row wire:key="row-empty">
<x-table.empty_cell colspan="16"></x-table.empty_cell>
</x-table.row>
@endforelse
```
good luck!

**Reply by Rashuoo7** [reply]
already did this.
`@forelse($courses as $course)
                <x-table.row wire:key="{{$course->id}}">
                    <x-table.cell>
                        <div class="text-sm font-medium text-gray-900">
                            <a href="{{route('admin.courses.show', ['course' => $course->id])}}">
                                {{ $course->title }}
                            </a>
                        </div>
                    </x-table.cell>
                    <x-table.cell>
                    {{--Edit Course Button--}}
                    <div x-data="{ open:false }" class="w-20 font-medium">
                        <a wire:click="edit({{$course->id}})"  href="#" @click="open = !open" class="text-yellow-500 hover:text-yellow-600">Edit</a>
                    </div>
                        {{-- /Edit Course Button--}}
                    </x-table.cell>
                    <x-table.cell>
                        <a href="#" wire:click="delete({{$course->id}})" class=" font-medium text-red-600 hover:text-red-900">Delete</a>
                    </x-table.cell>
                </x-table.row>
            @empty
                <x-table.row wire:key="row-empty">
                    <x-table.empty_cell colspan="16"></x-table.empty_cell>
                </x-table.row>
          @endforelse`

**Comment by gofish543** [comment]
This is an issue I faced as well and I tried `wire:key` but it didn't work either

**Reply by joshhanley** [reply]
@gofish543 Make sure to update to 3.0.1 as that should fix the issue.
